pkg/process/checks/process_windows_test.go

```diff
@@ -0,0 +1,44 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build windows
+// +build windows
+
+package checks
+
+import (
+	"sync"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/DataDog/datadog-agent/pkg/config"
+)
+
+func TestPerfCountersConfigSetting(t *testing.T) {
+	resetOnce := func() {
+		processProbeOnce = sync.Once{}
+	}
+
+	t.Run("use toolhelp API", func(t *testing.T) {
+		resetOnce()
+		defer resetOnce()
+
+		cfg := config.Mock()
+		cfg.Set("process_config.windows.use_perf_counters", false)
+		probe := getProcessProbe()
+		assert.Equal(t, probe, defaultWindowsProbe)
```

Comment:
nit: you can probably assert that you're able to cast `probe` to `windowsToolhelpProbe` in the first test case 
https://github.com/DataDog/datadog-agent/blob/b5bed4dc6a5b08a50346e614b386ea67e37695de/pkg/process/procutil/process_windows_toolhelp.go#L66-L68

and to `probe` in the second one
https://github.com/DataDog/datadog-agent/blob/b5bed4dc6a5b08a50346e614b386ea67e37695de/pkg/process/procutil/process_windows.go#L49-L58

Reply:
Good idea. I used `assert.IsType` to fix it